### PR TITLE
cpu: Fix segment fault when using debug flags Branch

### PR DESCRIPTION
**Pull request by rogerchang23424**
Change-Id: I36624b93f53aa101a57d51f3b917696cb2809136

### CODE REVIEW

File: src/cpu/pred/bpred_unit.cc [hunk 1]
```diff
@@ -503,10 +503,12 @@ BPredUnit::squash(const InstSeqNum &squashed_sn,
...
-                DPRINTF(Branch,
-                        "[tid:%i] [squash sn:%llu] Restoring top of RAS "
-                        "to: %i, target: %s\n", tid, squashed_sn,
-                        hist_it->RASIndex, *hist_it->RASTarget);
+                if (hist_it->RASTarget != nullptr) {
+                    DPRINTF(Branch,
+                            "[tid:%i] [squash sn:%llu] Restoring top of RAS "
+                            "to: %i, target: %s\n", tid, squashed_sn,
+                            hist_it->RASIndex, *hist_it->RASTarget.get());
+                }
```
Comment (powerjg) [inline src/cpu/pred/bpred_unit.cc]: Out of curiosity, why not something like this? Doesn't it make sense to print even if there is no `RASTarget`?

```suggestion
                 DPRINTF(Branch,
                        "[tid:%i] [squash sn:%llu] Restoring top of RAS "
                        "to: %i, target: %s\n", tid, squashed_sn,
                        hist_it->RASIndex, 
                        hist_it->RASIndex ? *hist_it->RASTarget.get() : "no RAS");
```

**Review by powerjg — COMMENTED** [review]
One small question

File: src/cpu/pred/bpred_unit.cc (hunk 1 above)
Comment (rogerchang23424) [reply src/cpu/pred/bpred_unit.cc]: Thanks for your suggestion. I'm not familiar with the part for cpu branch prediction, any changes make --debug-flags=Branch will LGTM.

**Review by powerjg — APPROVED** [review]
That was much more complicated than I expected! Thanks for the updated.

**Pull request merged**